extension/history.js

```diff
@@ -20,107 +21,124 @@ export class Database {
+      request.onerror = e => {
+        const code = e.target.errorCode;
+        reject(new Error(`Unable to retrieve from database: ${code}`));
+      };
     });
   }
 
   getAll(TBName) {
-    return new Promise(resolve => {
-      const request = indexedDB.open(this.DBName);
+    return new Promise((resolve, reject) => {
+      const request = indexedDB.open(this.dbName);
       request.onsuccess = e => {
         const database = e.target.result;
         const objectStore = database.transaction(TBName).objectStore(TBName);
         const list = [];
-
-        objectStore.openCursor().onsuccess = event => {
-          const cursor = event.target.result;
+        // return all objects from the object store according to `this.order`
+        const request = objectStore.openCursor(null, this.order);
+        request.onsuccess = e => {
+          const cursor = e.target.result;
           if (cursor) {
             list.push(cursor.value);
             cursor.continue();
           } else {
             resolve(list);
           }
         };
+        request.onerror = e => {
+          const code = e.target.errorCode;
+          reject(new Error(`Unable to retrieve data from database: ${code}`));
+        };
       };
     });
   }
 
   add(obj, TBName) {
     return new Promise((resolve, reject) => {
-      const request = indexedDB.open(this.DBName);
+      const request = indexedDB.open(this.dbName);
       request.onsuccess = e => {
         const database = e.target.result;
         const add = database
           .transaction([TBName], this.readwrite)
           .objectStore(TBName)
           .add(obj);
-        add.onsuccess = event => {
-          // log.info("1 record has been added to your database.");
-          resolve("1 record has been added to your database.");
+        add.onsuccess = e => {
+          log.info("1 record has been added to your database.");
+          resolve();
+        };
+        add.onerror = e => {
+          const code = e.target.errorCode;
+          reject(new Error(`Unable to add data records: ${code}`));
         };
-        add.onerror = event => reject("Unable to add data records");
         database.close();
       };
     });
   }
 
   delete(primaryKey, TBName) {
     return new Promise((resolve, reject) => {
-      const request = indexedDB.open(this.DBName);
+      const request = indexedDB.open(this.dbName);
       request.onsuccess = e => {
         const database = e.target.result;
         const remove = database
           .transaction([TBName], this.readwrite)
           .objectStore(TBName)
           .delete(primaryKey);
-        remove.onerror = event =>
-          reject("This entry could not be removed from database.");
-        remove.onsuccess = event =>
-          resolve("This entry has been removed from your database.");
+        remove.onsuccess = e => {
+          log.info("1 record has been added to your database.");
+          resolve();
```

Comment:
I believe this should log `This entry has been removed from your database`.

Reply:
Fixed!